docs/tutorials/developer-journey/level-5/5.2-ICP-ETH-tutorial.md

```diff
@@ -0,0 +1,433 @@
+    await core.addNfts(caller, nfts);
+  };
+
+  /// Define a method to retrieve the NFTs which were previously verified via `addNfts()`:
+  public query ({ caller }) func getNfts() : async [Types.Nft.Nft] {
+    core.getNfts(caller);
+  };
+
+  /// Define a method to hide or show all NFTs with the given smart contract or wallet address:
+  public shared ({ caller }) func setAddressFiltered(address : Types.Address.Address, filtered : Bool) : async Bool {
+    await core.setAddressFiltered(caller, address, filtered);
+  };
+
+  /// Define a method to retrieve the full log for this canister:
+  public query ({ caller }) func getHistory() : async ?[History.Event] {
+    core.getHistory(caller);
+  };
+
+  /// Define a method to retrieve a public-facing log for this canister:
+  public query ({ caller }) func getPublicHistory() : async [Types.PublicEvent] {
+    core.getPublicHistory(caller);
+  };
+
+  /// Define a method to retrieve a Candid representation of the canister's internal state:
+  public query ({ caller }) func getState() : async ?[Snapshot.Entry] {
+    core.getState(caller);
+  };
+
+};
+```
+
+Next, take a look at the Candid file for the `ic_eth` canister. Recall that Candid is used to describe the interfaces of a canister. The Candid file for the project's `ic_eth` canister will provide insight into the functions defined within the canister that are responsible for the dapp's interactions with the Ethereum network:
+
+```candid
+service : {
+  verify_ecdsa : (eth_address : text, message : text, signature : text) -> (bool) query;
+  erc721_owner_of : (network : text, contract_address : text, token_id : nat64) -> (text);
+  erc1155_balance_of : (network : text, contract_address : text, owner_address : text, token_id : nat64) -> (nat);
+}
+```
+
+In this Candid file, you can see that there are three services defined:
+
+- `verify_ecdsa`: This service takes input of an `eth_address`, `message`, and `signature`, all of type `text` and returns a query call of type `bool`. Type `bool` instances can have the value of either 'True' or 'False'. 
+
+- `erc721_owner_of`: This service takes input of `network` and `contract_address` of type `text`, and `token_id` of type `nat64`, and returns a `text` value. 
+
+- `erc1155_balance_of`: This service takes input of `network`, `contract_address`, and `owner_address` of type `text`, and `token_id` of type `nat64`, and returns a `nat` value. 
+
+This tutorial will not dive into the frontend's configuration, as it is focused on the vetKey implementation within the backend of the dapp. If you want to learn more about frontend canisters, check out the documentation [here](/docs/current/developer-docs/frontend).
+
+### Creating a testnet wallet
+
+Before you can deploy this starter project, you will need an Ethereum testnet wallet. To get one, first install [MetaMask](https://metamask.io/). MetaMask is supported as a browser extension for most web browsers. 
```

Comment:
Nit: You can deploy the starter project but you can't test the functionality as a user without an Ethereum wallet and some (testnet) ETH.

Reply:
Revised to include not regarding testing functionality